plugins/obs-ffmpeg/vaapi-utils.c

```diff
@@ -0,0 +1,263 @@
+// SPDX-FileCopyrightText: 2022 tytan652 <tytan652@tytanium.xyz>
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "vaapi-utils.h"
+
+#include <util/bmem.h>
+#include <util/dstr.h>
+
+#include <va/va_drm.h>
+#include <va/va_str.h>
+
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+static bool version_logged = false;
+static int drm_fd = -1;
+
+static const char *default_h264_device = NULL;
+
+static VADisplay vaapi_open_display_drm(const char *device_path)
+{
+	VADisplay va_dpy;
+
+	if (!device_path)
+		return NULL;
+
+	drm_fd = open(device_path, O_RDWR);
```

Comment:
`vaapi_open` is being called from callbacks, id prefer if we actually handle the `drm_fd` in a reentrant way.

Reply:
Changes done waiting for your re-review before squashing.